### PR TITLE
Feature/70 rework related items post modules

### LINKED ISSUE #70: Rework the UI/Code for Post Modules to select related items
Based on the ACF implementation adapt Integration 

- [x] add Pods as data-source option, remove the current selector
- [x] make sure the old settings still work until a user updates the module setting

### PR DESCRIPTION
**Pull request by quasel**
fixes #70

### CODE REVIEW

File: pods-beaver-themer.php [hunk 1]
```diff
@@ -407,12 +410,67 @@ function pods_beaver_render_settings_field_order_by( $field, $name, $settings )
...
+ * @param int $post_id The ID of the post
+ *
+ * @return mixed
+ */
+function pods_beaver_update_module_settings_data_source( $data, $status, $post_id ) {
+
+	foreach ( $data as $node ) {
+		if ( 'module' === $node->type && property_exists( $node->settings, 'use_pods' ) ) {
+			$module_settings = $node->settings;
+			if ( ! ( 'no' === $module_settings->use_pods ) ) {
```
Comment (sc0ttkclark) [inline pods-beaver-themer.php]: `'no' !== $module_settings->use_pods` instead

File: pods-beaver-themer.php [hunk 2]
```diff
@@ -173,29 +176,18 @@ function pods_beaver_loop_settings_before_form( $settings ) {
 	}
 
 	$setting_fields = array(
-		'use_pods'                      => array(
+		'pods_source_type'                      => array(
```
Comment (sc0ttkclark) [inline pods-beaver-themer.php]: Does changing the setting field name mean the settings won't match up with previous installs? Backcompat issues here?

Comment (quasel) [reply pods-beaver-themer.php]: that's taken care of with `pods_beaver_update_module_settings_data_source()`  the filter runs straight after any DB call to get settings, so the correct values are cached / and saved if edited 
this maybe can be improved later on once BB releases 2.2 in a few months.

File: pods-beaver-themer.php (hunk 1 above)
Comment (quasel) [reply pods-beaver-themer.php]: changed

File: pods-beaver-themer.php (hunk 2 above)
Comment (sc0ttkclark) [reply pods-beaver-themer.php]: But that doesn't handle initial load or populating the setting field? Seems like it could still break things.

Comment (quasel) [reply pods-beaver-themer.php]: it odes it's the single point where BB get's the information out of the DB got the tip from Justin (Lead Dev @BB)

Comment (quasel) [reply pods-beaver-themer.php]: https://www.dropbox.com/s/zgw1w5sma3bfga5/Screenshot%202018-06-19%2023.07.34.png?dl=0

Comment (quasel) [reply pods-beaver-themer.php]: Modules will have a `filter_settings` method in 2.2. but that's at least 2 months out

Comment (sc0ttkclark) [reply pods-beaver-themer.php]: Gotcha, was just confused, it looked like it was only happening when DB was updated.

File: pods-beaver-themer.php [hunk 3]
```diff
@@ -398,21 +404,77 @@ function pods_beaver_render_settings_field_order_by( $field, $name, $settings )
...
+ * @param object $settings
+ *
+ * @return array
+ */
+function pods_beaver_render_settings_field( $field, $name, $settings ) {
+	if ( 'data_source' != $name ) {
+		return $field;
+	}
 
+	$field['options']['pods_relationship'] = __( 'PODS Relationship', 'pods-beaver-builder-themer-add-on' );
```
Comment (sc0ttkclark) [inline pods-beaver-themer.php]: Can you fix the caps on PODS to be Pods?

Comment (quasel) [reply pods-beaver-themer.php]: yes will go over all labels before the Release

**Pull request merged**